.github/workflows/Bonsai.yml

```diff
@@ -0,0 +1,460 @@
+        run: dotnet build Bonsai.sln --no-restore --configuration ${{matrix.configuration}}
+
+      # ----------------------------------------------------------------------- Repack bootstrapper
+      # This happens before pack since the bootstrapper package uses it
+      - name: Repack bootstrapper
+        if: matrix.create-installer || env.UseRepackForBootstrapperPackage == 'true'
+        run: dotnet build Bonsai --no-restore --configuration ${{matrix.configuration}} -t:Repack
+
+      # ----------------------------------------------------------------------- Pack
+      # Since packages are core to Bonsai functionality we always pack them even if they won't be collected
+      - name: Pack
+        id: pack
+        run: dotnet pack Bonsai.sln --no-restore --no-build --configuration ${{matrix.configuration}}
+
+      # ----------------------------------------------------------------------- Test
+      - name: Test
+        if: '!matrix.dummy-build'
+        run: dotnet test Bonsai.sln --no-restore --no-build --configuration ${{matrix.configuration}} --verbosity normal
+
+      # ----------------------------------------------------------------------- Create portable zip
+      - name: Create portable zip
+        id: create-portable-zip
+        if: matrix.create-installer
+        run: python .github/workflows/create-portable-zip.py artifacts/Bonsai.zip ${{matrix.configuration}}
+        env:
+          NUGET_API_URL: ${{vars.NUGET_API_URL}}
+          # This should be kept in sync with publish-packages-nuget-org
+          IS_FULL_RELEASE: ${{github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && github.event.inputs.will_publish_packages == 'true' && github.event.inputs.version != '')}}
+
+      # ----------------------------------------------------------------------- Build Visual Studio templates
+      - name: Build Visual Studio templates
+        if: matrix.create-installer
+        run: msbuild /nologo /maxCpuCount Bonsai.Templates/Bonsai.Templates.sln /p:Configuration=${{matrix.configuration}} /p:DeployExtension=false
+
+      # ----------------------------------------------------------------------- Build setup
+      - name: Restore setup
+        if: matrix.create-installer
+        # Restoring the packages.config directly means we don't rely on the system-wide Wix install
+        run: |
+          nuget restore Bonsai.Setup/packages.config -SolutionDir .
+          nuget restore Bonsai.Setup.Bootstrapper/packages.config -SolutionDir .
+
+      - name: Build Setup
+        id: create-installer
+        if: matrix.create-installer
+        run: msbuild /nologo /maxCpuCount Bonsai.Setup.sln /p:Configuration=${{matrix.configuration}}
+      
+      # ----------------------------------------------------------------------- Collect artifacts
+      - name: Collect NuGet packages
+        uses: actions/upload-artifact@v4
+        if: matrix.collect-packages && steps.pack.outcome == 'success' && always()
```

Comment:
The main reason for it is that it allows this step to run even if previous steps failed.

The main idea is that if a build is partially failing for some reason that you still get access to whatever artifacts did manage to build before the failure, which is frequently useful for debugging and is helpful to someone who wants some of the artifacts.

For example: If building the Visual Studio templates failed for some reason, you'd still have access to the portable zip.

To frame it another way: `Collecting NuGet Packages` really only cares that `collect-packages` was successful, so it will always run as long as it's successful (even if other things failed.)

Under the hood, every single `if` in GitHub Actions has an implicit `success() &&` prefix. Adding [things like `always()`](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions) anywhere to the expression is disabling that behavior.